internal/repository/sync_test.go

```diff
@@ -0,0 +1,493 @@
+	if err != nil {
+		t.Fatal(err)
+	}
+	targetsKeyBytes, err := os.ReadFile(filepath.Join("test-data", "targets"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := remoteRepo.InitializeRoot(context.Background(), rootKeyBytes, false); err != nil {
+		t.Fatal(err)
+	}
+	if err := remoteRepo.AddTopLevelTargetsKey(context.Background(), rootKeyBytes, targetsPubKeyBytes, false); err != nil {
+		t.Fatal(err)
+	}
+	if err := remoteRepo.InitializeTargets(context.Background(), targetsKeyBytes, policy.TargetsRoleName, false); err != nil {
+		t.Fatal(err)
+	}
+	emptyTreeHash, err := gitinterface.WriteTree(remoteRepo.r, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	refName := "refs/heads/main"
+	anotherRefName := "refs/heads/feature"
+	commitID, err := gitinterface.Commit(remoteRepo.r, emptyTreeHash, refName, "Initial commit", false)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := remoteRepo.RecordRSLEntryForReference(refName, false); err != nil {
+		t.Fatal(err)
+	}
+	if err := remoteRepo.r.Storer.SetReference(plumbing.NewSymbolicReference(plumbing.HEAD, plumbing.ReferenceName(refName))); err != nil {
+		t.Fatal(err)
+	}
+	if err := remoteRepo.r.Storer.SetReference(plumbing.NewHashReference(plumbing.ReferenceName(anotherRefName), commitID)); err != nil {
+		t.Fatal(err)
+	}
+	if err := remoteRepo.RecordRSLEntryForReference(anotherRefName, false); err != nil {
+		t.Fatal(err)
+	}
+
+	currentDir, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	remoteRSLRef, err := remoteRepo.r.Reference(plumbing.ReferenceName(rsl.RSLRef), true)
+	if err != nil {
+		t.Fatal(err)
+	}
+	remotePolicyRef, err := remoteRepo.r.Reference(plumbing.ReferenceName(policy.PolicyRef), true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	t.Run("successful clone without specifying dir", func(t *testing.T) {
+		localTmpDir, err := os.MkdirTemp("", "gittuf")
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer os.RemoveAll(localTmpDir) //nolint:errcheck
+
+		if err := os.Chdir(localTmpDir); err != nil {
+			t.Fatal(err)
+		}
+		defer os.Chdir(currentDir) //nolint:errcheck
+
+		repo, err := Clone(context.Background(), remoteTmpDir, "", "")
+		assert.Nil(t, err)
+		head, err := repo.r.Head()
+		if err != nil {
+			t.Fatal(err)
+		}
+		assert.Equal(t, commitID, head.Hash())
+
+		localRSLRef, err := repo.r.Reference(plumbing.ReferenceName(rsl.RSLRef), true)
+		if err != nil {
+			t.Fatal(err)
+		}
+		assert.Equal(t, remoteRSLRef.Hash(), localRSLRef.Hash())
+		localPolicyRef, err := repo.r.Reference(plumbing.ReferenceName(policy.PolicyRef), true)
+		if err != nil {
+			t.Fatal(err)
+		}
+		assert.Equal(t, remotePolicyRef.Hash(), localPolicyRef.Hash())
+	})
+
+	t.Run("successful clone with dir", func(t *testing.T) {
+		localTmpDir, err := os.MkdirTemp("", "gittuf")
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer os.RemoveAll(localTmpDir) //nolint:errcheck
+
+		if err := os.Chdir(localTmpDir); err != nil {
+			t.Fatal(err)
+		}
+		defer os.Chdir(currentDir) //nolint:errcheck
```

Comment:
This is repeated a lot. Pull into a helper func.

Reply:
I think I cannot? Because the deferred chdir to currentDir will trigger when exiting the helper instead of the test.